src/sqlfluff/rules/L043.py

```diff
@@ -216,11 +216,18 @@ def _eval(self, context: RuleContext) -> Optional[LintResult]:
 
                 # Locate column reference in condition expression.
                 column_reference_segment = next(
-                    segment
-                    for segment in condition_expression.segments
-                    if segment.type == "column_reference"
+                    (
+                        segment
+                        for segment in condition_expression.segments
+                        if segment.type == "column_reference"
+                    ),
+                    None,
                 )
 
+                # Return None if no column reference is detected (this condition does not apply to funcitons).
```

Comment:
```suggestion
                # Return None if no column reference is detected (this condition does not apply to functions).
```

Reply:
actually "funcitons" is the hot new programming paradigm 😛

Reply:
fixed it now 😄